src/model_extensions/custom_visualization.py

```diff
@@ -1,54 +1,113 @@
-from sodym.tools.config import cfg
-from sodym.tools.visualize.vistools import show_and_save_pyplot, visualize_if_set_in_config
-from sodym.tools.visualize.plot_array import visualize_array
+import logging
+import os
+
+from matplotlib import pyplot as plt
+import plotly.graph_objects as go
+
 from sodym.classes.mfa_system import MFASystem
+from sodym.export.data_writer import DataWriter
+from sodym.export.helper import ArrayPlotter
+
+
+class CustomDataVisualizer(DataWriter):
```

Comment:
Maybe call it Exporter? If it not only visualizes, but also exports?
Also here, it may be easier to separate the two tasks into two classes?